test/ERC6538Registry.t.sol

```diff
@@ -1,20 +1,19 @@
-// SPDX-License-Identifier: UNLICENSED
-pragma solidity 0.8.20;
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
 
 import {Test} from "forge-std/Test.sol";
 import {Deploy} from "script/Deploy.s.sol";
 
 contract ERC6538RegistryTest is Test, Deploy {
+  SigUtils internal sigUtils;
+
   event StealthMetaAddressSet(
-    bytes indexed registrant, uint256 indexed schemeId, bytes stealthMetaAddress
+    address indexed registrant, uint256 indexed schemeId, bytes stealthMetaAddress
   );
 
   function setUp() public {
     Deploy.run();
-  }
-
-  function toBytes(address who) internal pure returns (bytes memory) {
-    return bytes.concat(bytes32(uint256(uint160(who))));
+    sigUtils = new SigUtils(registry.DOMAIN_SEPARATOR());
   }
 }
 
```

Comment:
*arbitrary comment location*

Should we have a unit test or two for the constructor showing the chainId and domain separator are properly initialized?

Reply:
Since `INITIAL_CHAIN_ID` and `INITIAL_DOMAIN_SEPARATOR` are immutable variables and baked into the bytecode and don't have a storage slot, I don't think there's an easy way to read them easily using `vm.load`. Maybe @mds1 knows a way around this. But for now, I've added a test to check that the `DOMAIN_SEPARATOR` is initialized correctly. It might be good way to indirectly test both `INITIAL_CHAIN_ID` and `INITIAL_DOMAIN_SEPARATOR` are set from the get go.

Reply:
Sounds good! Pull out this for us to consider revisiting it soon https://github.com/ScopeLift/stealth-address-erc-contracts/issues/6